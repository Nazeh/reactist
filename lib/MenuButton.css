.reactist.menu_button .body_wrapper .body {
  width: 100%;
  overflow: auto;
}
.reactist.menu_button .with_arrow:before,
.reactist.menu_button .with_arrow:after {
  display: none !important;
}
.reactist.menu_button .menu_button_trigger {
  background-color: white;
  user-select: none;
  font-size: 0.875rem;
  color: #808080;
  font-weight: 400;
  line-height: 1.7;
  display: inline-block;
  min-width: 100px;
  border: 1px solid #ececec;
  padding: 5px 10px;
  border-radius: 4px;
  text-align: center;
}
.reactist.menu_button .menu_button_trigger:hover {
  cursor: pointer;
}
.reactist.menu_button .menu_button_items .menu_button_item {
  padding: 5px 10px;
  white-space: nowrap;
}
.reactist.menu_button .menu_button_items .menu_button_item .name {
  font-size: 0.875rem;
  color: #202020;
  font-weight: 400;
  line-height: 1.7;
  margin: 0;
}
.reactist.menu_button .menu_button_items .menu_button_item:hover {
  background-color: #ececec;
  cursor: pointer;
}
.reactist.menu_button .menu_button_items .menu_button_item:nth-child(2) {
  border-top: 1px solid #ececec;
  border-bottom: 1px solid #ececec;
}

.reactist.dropdown .trigger {
  cursor: pointer;
}
.reactist.dropdown .body {
  border-radius: 3px;
  border: 1px solid #dcdcdc;
  overflow: hidden;
  box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.08);
  z-index: 1;
  background-color: #fff;
}
.reactist.dropdown hr {
  border: none;
  height: 1px;
  background-color: #ececec;
  margin: 0 5px;
}
.reactist.dropdown .with_arrow:after,
.reactist.dropdown .with_arrow:before {
  z-index: 1;
  content: '';
  display: block;
  position: absolute;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 6px;
  right: 14px;
}
.reactist.dropdown .with_arrow:after {
  top: -11px;
  border-color: transparent transparent #fff transparent;
}
.reactist.dropdown .with_arrow:before {
  top: -12px;
  border-color: transparent transparent #dcdcdc transparent;
}
.reactist.dropdown .with_arrow.top:after {
  top: -1px;
  border-color: #fff transparent transparent transparent;
}
.reactist.dropdown .with_arrow.top:before {
  top: -1px;
  right: 13px;
  border-width: 7px;
  border-color: #dcdcdc transparent transparent transparent;
}


/*# sourceMappingURL=MenuButton.map*/