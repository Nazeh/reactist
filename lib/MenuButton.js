!function(t,e){"object"==typeof exports&&"object"==typeof module?module.exports=e(require("prop-types"),require("react"),require("classnames"),require("react-dom")):"function"==typeof define&&define.amd?define("reactist",["prop-types","react","classnames","react-dom"],e):"object"==typeof exports?exports.reactist=e(require("prop-types"),require("react"),require("classnames"),require("react-dom")):t.reactist=e(t["prop-types"],t.React,t.classnames,t.ReactDOM)}(window,function(t,e,n,r){return function(t){var e={};function n(r){if(e[r])return e[r].exports;var o=e[r]={i:r,l:!1,exports:{}};return t[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=t,n.c=e,n.d=function(t,e,r){n.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:r})},n.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},n.t=function(t,e){if(1&e&&(t=n(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var o in t)n.d(r,o,function(e){return t[e]}.bind(null,o));return r},n.n=function(t){var e=t&&t.__esModule?function(){return t.default}:function(){return t};return n.d(e,"a",e),e},n.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},n.p="",n(n.s=42)}({0:function(e,n){e.exports=t},1:function(t,n){t.exports=e},11:function(t,e,n){var r=n(12);"string"==typeof r&&(r=[[t.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(t.exports=r.locals)},12:function(t,e,n){(t.exports=n(3)(!1)).push([t.i,".reactist.dropdown .trigger {\n  cursor: pointer;\n}\n.reactist.dropdown .body {\n  border-radius: 3px;\n  border: 1px solid #dcdcdc;\n  overflow: hidden;\n  box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.08);\n  z-index: 1;\n  background-color: #fff;\n}\n.reactist.dropdown hr {\n  border: none;\n  height: 1px;\n  background-color: #ececec;\n  margin: 0 5px;\n}\n.reactist.dropdown .with_arrow:after,\n.reactist.dropdown .with_arrow:before {\n  z-index: 1;\n  content: '';\n  display: block;\n  position: absolute;\n  width: 0;\n  height: 0;\n  border-style: solid;\n  border-width: 6px;\n  right: 14px;\n}\n.reactist.dropdown .with_arrow:after {\n  top: -11px;\n  border-color: transparent transparent #fff transparent;\n}\n.reactist.dropdown .with_arrow:before {\n  top: -12px;\n  border-color: transparent transparent #dcdcdc transparent;\n}\n.reactist.dropdown .with_arrow.top:after {\n  top: -1px;\n  border-color: #fff transparent transparent transparent;\n}\n.reactist.dropdown .with_arrow.top:before {\n  top: -1px;\n  right: 13px;\n  border-width: 7px;\n  border-color: #dcdcdc transparent transparent transparent;\n}\n",""])},2:function(t,e){t.exports=n},3:function(t,e){t.exports=function(t){var e=[];return e.toString=function(){return this.map(function(e){var n=function(t,e){var n=t[1]||"",r=t[3];if(!r)return n;if(e&&"function"==typeof btoa){var o=function(t){return"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(t))))+" */"}(r),i=r.sources.map(function(t){return"/*# sourceURL="+r.sourceRoot+t+" */"});return[n].concat(i).concat([o]).join("\n")}return[n].join("\n")}(e,t);return e[2]?"@media "+e[2]+"{"+n+"}":n}).join("")},e.i=function(t,n){"string"==typeof t&&(t=[[null,t,""]]);for(var r={},o=0;o<this.length;o++){var i=this[o][0];"number"==typeof i&&(r[i]=!0)}for(o=0;o<t.length;o++){var a=t[o];"number"==typeof a[0]&&r[a[0]]||(n&&!a[2]?a[2]=n:n&&(a[2]="("+a[2]+") and ("+n+")"),e.push(a))}},e}},4:function(t,e,n){var r={},o=function(t){var e;return function(){return void 0===e&&(e=t.apply(this,arguments)),e}}(function(){return window&&document&&document.all&&!window.atob}),i=function(t){var e={};return function(t,n){if("function"==typeof t)return t();if(void 0===e[t]){var r=function(t,e){return e?e.querySelector(t):document.querySelector(t)}.call(this,t,n);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(t){r=null}e[t]=r}return e[t]}}(),a=null,s=0,c=[],u=n(5);function l(t,e){for(var n=0;n<t.length;n++){var o=t[n],i=r[o.id];if(i){i.refs++;for(var a=0;a<i.parts.length;a++)i.parts[a](o.parts[a]);for(;a<o.parts.length;a++)i.parts.push(b(o.parts[a],e))}else{var s=[];for(a=0;a<o.parts.length;a++)s.push(b(o.parts[a],e));r[o.id]={id:o.id,refs:1,parts:s}}}}function p(t,e){for(var n=[],r={},o=0;o<t.length;o++){var i=t[o],a=e.base?i[0]+e.base:i[0],s={css:i[1],media:i[2],sourceMap:i[3]};r[a]?r[a].parts.push(s):n.push(r[a]={id:a,parts:[s]})}return n}function d(t,e){var n=i(t.insertInto);if(!n)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=c[c.length-1];if("top"===t.insertAt)r?r.nextSibling?n.insertBefore(e,r.nextSibling):n.appendChild(e):n.insertBefore(e,n.firstChild),c.push(e);else if("bottom"===t.insertAt)n.appendChild(e);else{if("object"!=typeof t.insertAt||!t.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=i(t.insertAt.before,n);n.insertBefore(e,o)}}function f(t){if(null===t.parentNode)return!1;t.parentNode.removeChild(t);var e=c.indexOf(t);e>=0&&c.splice(e,1)}function h(t){var e=document.createElement("style");if(void 0===t.attrs.type&&(t.attrs.type="text/css"),void 0===t.attrs.nonce){var r=function(){0;return n.nc}();r&&(t.attrs.nonce=r)}return m(e,t.attrs),d(t,e),e}function m(t,e){Object.keys(e).forEach(function(n){t.setAttribute(n,e[n])})}function b(t,e){var n,r,o,i;if(e.transform&&t.css){if(!(i=e.transform(t.css)))return function(){};t.css=i}if(e.singleton){var c=s++;n=a||(a=h(e)),r=v.bind(null,n,c,!1),o=v.bind(null,n,c,!0)}else t.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(n=function(t){var e=document.createElement("link");return void 0===t.attrs.type&&(t.attrs.type="text/css"),t.attrs.rel="stylesheet",m(e,t.attrs),d(t,e),e}(e),r=function(t,e,n){var r=n.css,o=n.sourceMap,i=void 0===e.convertToAbsoluteUrls&&o;(e.convertToAbsoluteUrls||i)&&(r=u(r));o&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var a=new Blob([r],{type:"text/css"}),s=t.href;t.href=URL.createObjectURL(a),s&&URL.revokeObjectURL(s)}.bind(null,n,e),o=function(){f(n),n.href&&URL.revokeObjectURL(n.href)}):(n=h(e),r=function(t,e){var n=e.css,r=e.media;r&&t.setAttribute("media",r);if(t.styleSheet)t.styleSheet.cssText=n;else{for(;t.firstChild;)t.removeChild(t.firstChild);t.appendChild(document.createTextNode(n))}}.bind(null,n),o=function(){f(n)});return r(t),function(e){if(e){if(e.css===t.css&&e.media===t.media&&e.sourceMap===t.sourceMap)return;r(t=e)}else o()}}t.exports=function(t,e){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(e=e||{}).attrs="object"==typeof e.attrs?e.attrs:{},e.singleton||"boolean"==typeof e.singleton||(e.singleton=o()),e.insertInto||(e.insertInto="head"),e.insertAt||(e.insertAt="bottom");var n=p(t,e);return l(n,e),function(t){for(var o=[],i=0;i<n.length;i++){var a=n[i];(s=r[a.id]).refs--,o.push(s)}t&&l(p(t,e),e);for(i=0;i<o.length;i++){var s;if(0===(s=o[i]).refs){for(var c=0;c<s.parts.length;c++)s.parts[c]();delete r[s.id]}}}};var y=function(){var t=[];return function(e,n){return t[e]=n,t.filter(Boolean).join("\n")}}();function v(t,e,n,r){var o=n?"":r.css;if(t.styleSheet)t.styleSheet.cssText=y(e,o);else{var i=document.createTextNode(o),a=t.childNodes;a[e]&&t.removeChild(a[e]),a.length?t.insertBefore(i,a[e]):t.appendChild(i)}}},42:function(t,e,n){"use strict";n.r(e),n.d(e,"MenuButton",function(){return l}),n.d(e,"MenuButtonItem",function(){return p});n(43);var r=n(1),o=n.n(r),i=n(0),a=n.n(i),s=n(2),c=n.n(s),u=n(7),l=function(t){var e=t.className,n=t.name,r=t.onClick,i=t.children,a=c()("reactist menu_button",e);return i&&0!==i.length?o.a.createElement(u.default.Box,{className:a},o.a.createElement(u.default.Trigger,null,o.a.createElement("span",{className:"reactist menu_button_trigger"},n)),o.a.createElement(u.default.Body,null,o.a.createElement("div",{className:"reactist menu_button_items"},i))):o.a.createElement("div",{className:a,onClick:r},o.a.createElement("span",{className:"reactist menu_button_trigger"},n))};l.displayName="MenuButton",l.propTypes={name:a.a.string,onClick:a.a.func,children:a.a.arrayOf(a.a.node),className:a.a.string};var p=function(t){var e=t.className,n=t.onClick,r=t.children,i=c()("reactist menu_button_item",e);return o.a.createElement("div",{className:i,onClick:n},r)};p.displayName="MenuButtonItem",p.propTypes={onClick:a.a.func,children:a.a.oneOfType([a.a.string,a.a.node,a.a.arrayOf(a.a.node)]),className:a.a.string}},43:function(t,e,n){var r=n(44);"string"==typeof r&&(r=[[t.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(t.exports=r.locals)},44:function(t,e,n){(t.exports=n(3)(!1)).push([t.i,".reactist.menu_button .body_wrapper .body {\n  width: 100%;\n  overflow: auto;\n}\n.reactist.menu_button .with_arrow:before,\n.reactist.menu_button .with_arrow:after {\n  display: none !important;\n}\n.reactist.menu_button .menu_button_trigger {\n  background-color: white;\n  user-select: none;\n  font-size: 0.875rem;\n  color: #808080;\n  font-weight: 400;\n  line-height: 1.7;\n  display: inline-block;\n  min-width: 100px;\n  border: 1px solid #ececec;\n  padding: 5px 10px;\n  border-radius: 4px;\n  text-align: center;\n}\n.reactist.menu_button .menu_button_trigger:hover {\n  cursor: pointer;\n}\n.reactist.menu_button .menu_button_items .menu_button_item {\n  padding: 5px 10px;\n  white-space: nowrap;\n}\n.reactist.menu_button .menu_button_items .menu_button_item .name {\n  font-size: 0.875rem;\n  color: #202020;\n  font-weight: 400;\n  line-height: 1.7;\n  margin: 0;\n}\n.reactist.menu_button .menu_button_items .menu_button_item:hover {\n  background-color: #ececec;\n  cursor: pointer;\n}\n.reactist.menu_button .menu_button_items .menu_button_item:nth-child(2) {\n  border-top: 1px solid #ececec;\n  border-bottom: 1px solid #ececec;\n}\n",""])},5:function(t,e){t.exports=function(t){var e="undefined"!=typeof window&&window.location;if(!e)throw new Error("fixUrls requires window.location");if(!t||"string"!=typeof t)return t;var n=e.protocol+"//"+e.host,r=n+e.pathname.replace(/\/[^\/]*$/,"/");return t.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(t,e){var o,i=e.trim().replace(/^"(.*)"$/,function(t,e){return e}).replace(/^'(.*)'$/,function(t,e){return e});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i)?t:(o=0===i.indexOf("//")?i:0===i.indexOf("/")?n+i:r+i.replace(/^\.\//,""),"url("+JSON.stringify(o)+")")})}},6:function(t,e){t.exports=r},7:function(t,e,n){"use strict";n.r(e);n(11);var r=n(1),o=n.n(r),i=n(6),a=n.n(i),s=n(0),c=n.n(s),u=n(2),l=n.n(u);function p(t){return(p="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(t)}function d(t,e){if(!(t instanceof e))throw new TypeError("Cannot call a class as a function")}function f(t,e){for(var n=0;n<e.length;n++){var r=e[n];r.enumerable=r.enumerable||!1,r.configurable=!0,"value"in r&&(r.writable=!0),Object.defineProperty(t,r.key,r)}}function h(t,e,n){return e&&f(t.prototype,e),n&&f(t,n),t}function m(t,e){return!e||"object"!==p(e)&&"function"!=typeof e?g(t):e}function b(t){return(b=Object.setPrototypeOf?Object.getPrototypeOf:function(t){return t.__proto__||Object.getPrototypeOf(t)})(t)}function y(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Super expression must either be null or a function");t.prototype=Object.create(e&&e.prototype,{constructor:{value:t,writable:!0,configurable:!0}}),e&&v(t,e)}function v(t,e){return(v=Object.setPrototypeOf||function(t,e){return t.__proto__=e,t})(t,e)}function g(t){if(void 0===t)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return t}var w=function(t){function e(t,n){var r;return d(this,e),(r=m(this,b(e).call(this,t,n))).state={show_body:!1,top:t.top||!1},r._handleClickOutside=r._handleClickOutside.bind(g(g(r))),r._setPosition=r._setPosition.bind(g(g(r))),r._toggleShowBody=r._toggleShowBody.bind(g(g(r))),r._timeout=null,r}return y(e,o.a.Component),h(e,[{key:"componentWillUnmount",value:function(){document.removeEventListener("click",this._handleClickOutside,!0),this._timeout&&clearTimeout(this._timeout)}},{key:"_handleClickOutside",value:function(t){var e=this;a.a.findDOMNode(this).contains(t.target)?this.props.allowBodyInteractions||(this._timeout=setTimeout(function(){e.state.show_body&&e._toggleShowBody()},100)):this._toggleShowBody()}},{key:"_toggleShowBody",value:function(){this.state.show_body?(this.props.onHideBody&&this.props.onHideBody(),document.removeEventListener("click",this._handleClickOutside,!0)):(this.props.onShowBody&&this.props.onShowBody(),document.addEventListener("click",this._handleClickOutside,!0)),this.setState({show_body:!this.state.show_body})}},{key:"_getTriggerComponent",value:function(){var t=this.props.children[0];return o.a.cloneElement(t,{onClick:this._toggleShowBody})}},{key:"_setPosition",value:function(t){if(t){var e=document.getElementById(this.props.scrolling_parent);if(e){var n=a.a.findDOMNode(this),r=a.a.findDOMNode(this).offsetTop,o=n.querySelector(".trigger").clientHeight,i=t.clientHeight,s=e.clientHeight+e.scrollTop-r-o<i;s!==this.state.top&&this.setState({top:s})}}}},{key:"_getBodyComponent",value:function(){var t=this.state.top,e=this.props,n=e.right,r=void 0!==n&&n,i=e.children[1],a={top:t,right:r,setPosition:this._setPosition},s=l()({body_wrapper:!0,with_arrow:!0,top:t,bottom:!t});return!!this.state.show_body&&o.a.createElement("div",{className:s,style:{position:"relative"}},o.a.cloneElement(i,a))}},{key:"render",value:function(){var t=l()("reactist dropdown",this.props.className),e=this.state.top;return o.a.createElement("div",{style:{display:"inline-block"},className:t},e&&this._getBodyComponent(),this._getTriggerComponent(),!e&&this._getBodyComponent())}}]),e}();w.displayName="Dropdown.Box",w.propTypes={top:c.a.bool,right:c.a.bool,scrolling_parent:c.a.string,allowBodyInteractions:c.a.bool,onShowBody:c.a.func,onHideBody:c.a.func,className:c.a.string,children:c.a.any};var _=function(t){function e(t,n){var r;return d(this,e),(r=m(this,b(e).call(this,t,n)))._onClick=r._onClick.bind(g(g(r))),r}return y(e,o.a.Component),h(e,[{key:"_onClick",value:function(t){t.preventDefault(),t.stopPropagation(),this.props.onClick(t)}},{key:"render",value:function(){return o.a.createElement("div",{style:{display:"block"},className:"trigger",onClick:this._onClick},this.props.children)}}]),e}();_.displayName="Dropdown.Trigger",_.propTypes={onClick:c.a.func,children:c.a.any};var x=function(t){function e(){return d(this,e),m(this,b(e).apply(this,arguments))}return y(e,o.a.Component),h(e,[{key:"render",value:function(){var t={position:"absolute",right:0,top:0};return this.props.top&&(t.top="auto",t.bottom=0),this.props.right&&(t.right="auto",t.left=0),o.a.createElement("div",{ref:this.props.setPosition,style:t,className:"body",id:"reactist-dropdown-body"},this.props.children)}}]),e}();x.displayName="Dropdown.Body",x.propTypes={top:c.a.bool,right:c.a.bool,setPosition:c.a.func,children:c.a.any},e.default={Box:w,Trigger:_,Body:x}}})});
//# sourceMappingURL=MenuButton.map