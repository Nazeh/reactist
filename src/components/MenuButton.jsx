import './styles/menu_button.less'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Dropdown from './Dropdown'

const MenuButton = ({ className, name, onClick, children }) => {
    const menuButtonClass = classNames('reactist menu_button', className)

    if (!children || children.length === 0) {
        return (
            <div className={menuButtonClass} onClick={onClick}>
                <span className="reactist menu_button_trigger">{name}</span>
            </div>
        )
    }

    return (
        <Dropdown.Box className={menuButtonClass}>
            <Dropdown.Trigger>
                <span className="reactist menu_button_trigger">{name}</span>
            </Dropdown.Trigger>
            <Dropdown.Body>
                <div className="reactist menu_button_items">{children}</div>
            </Dropdown.Body>
        </Dropdown.Box>
    )
}
MenuButton.displayName = 'MenuButton'
MenuButton.propTypes = {
    /** Name that is displayed as trigger of the MenuButton.  */
    name: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
    /** When not providing any children you can control what happens when the MenuButton is clicked. */
    onClick: PropTypes.func,
    /** Children are displayed as items after clicking on the MenuButton. */
    children: PropTypes.arrayOf(PropTypes.node),
    /** Additional css class applied to the MenuButton. */
    className: PropTypes.string
}

const MenuButtonItem = ({ className, onClick, children }) => {
    const menuButtonItemClass = classNames(
        'reactist menu_button_item',
        className
    )
    return (
        <div className={menuButtonItemClass} onClick={onClick}>
            {children}
        </div>
    )
}
MenuButtonItem.displayName = 'MenuButtonItem'
MenuButtonItem.propTypes = {
    /** Callback when MenuButtonItem is clicked. */
    onClick: PropTypes.func,
    /** Content of the MenuButtonItem. */
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node)
    ]),
    /** Additional css class applied to the MenuButtonItem. */
    className: PropTypes.string
}

export { MenuButton, MenuButtonItem }
