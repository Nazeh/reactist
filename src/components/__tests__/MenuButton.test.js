import React from 'react'
import { shallow } from 'enzyme'

import { MenuButton, MenuButtonItem } from '../MenuButton'

describe('MenuButton', () => {
    describe('without items', () => {
        it('renders without items', () => {
            const menuButton = shallow(<MenuButton name="Test Menu" />)
            expect(menuButton).toMatchSnapshot()
        })

        it('calls onClick callback when rendering without items', () => {
            const onClickSpy = jest.fn()
            const menuButton = shallow(
                <MenuButton name="Test Menu" onClick={onClickSpy} />
            )

            menuButton.simulate('click')
            expect(onClickSpy).toHaveBeenCalled()
        })
    })

    describe('with items', () => {
        it('renders with items', () => {
            const menuButton = shallow(
                <MenuButton name="Test Menu">
                    <MenuButtonItem>Item 1</MenuButtonItem>
                    <MenuButtonItem>Item 2</MenuButtonItem>
                </MenuButton>
            )
            expect(menuButton).toMatchSnapshot()
        })

        it('calls onClick callbacks of items', () => {
            const onClickSpy1 = jest.fn()
            const onClickSpy2 = jest.fn()
            const menuButton = shallow(
                <MenuButton name="Test Menu">
                    <MenuButtonItem onClick={onClickSpy1}>
                        Item 1
                    </MenuButtonItem>
                    <MenuButtonItem onClick={onClickSpy2}>
                        Item 2
                    </MenuButtonItem>
                </MenuButton>
            )

            menuButton
                .find('MenuButtonItem')
                .get(0)
                .props.onClick()
            expect(onClickSpy1).toHaveBeenCalledTimes(1)
            expect(onClickSpy2).toHaveBeenCalledTimes(0)
            menuButton
                .find('MenuButtonItem')
                .get(1)
                .props.onClick()
            expect(onClickSpy1).toHaveBeenCalledTimes(1)
            expect(onClickSpy2).toHaveBeenCalledTimes(1)
        })

        it('can use item without menu', () => {
            const onClickSpy = jest.fn()
            const item = shallow(
                <MenuButtonItem onClick={onClickSpy}>Item 1</MenuButtonItem>
            )

            item.simulate('click')
            expect(onClickSpy).toHaveBeenCalled()
            expect(item.text()).toBe('Item 1')
        })
    })
})
