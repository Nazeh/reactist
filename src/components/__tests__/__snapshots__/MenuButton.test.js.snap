// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton with items renders with items 1`] = `
<Dropdown.Box
  className="reactist_menu_button"
>
  <Dropdown.Trigger>
    <span
      className="reactist_menu_button_trigger"
    >
      Test Menu
    </span>
  </Dropdown.Trigger>
  <Dropdown.Body>
    <div
      className="reactist_menu_button_items"
    >
      <MenuButtonItem>
        Item 1
      </MenuButtonItem>
      <MenuButtonItem>
        Item 2
      </MenuButtonItem>
    </div>
  </Dropdown.Body>
</Dropdown.Box>
`;

exports[`MenuButton without items renders without items 1`] = `
<div
  className="reactist_menu_button"
>
  <span
    className="reactist_menu_button_trigger"
  >
    Test Menu
  </span>
</div>
`;
